data-raw/pstr_data.R

```diff
@@ -3,7 +3,7 @@
 pstr_companies <- tibble::tribble(
   ~company_id,        ~company_name,               ~products,                      ~sector,                     ~subsector,
             1,      "Peasant Peter",                "screws", "steel_metal_transformation",    "bending_steels_and_metals",
-            2,      "Peasant Peter",       "aluminium_alloy",               "steel_metals",             "aluminium_alloys",
+            2,      "Peasant Peter",                      NA,               "steel_metals",             "aluminium_alloys",
             3,      "Peasant Peter",      "ultrafilstration",                       "cars", "vehicle_maintenance_products",
             4,       "Peasant Paul",           "frozen_food",                "frozen_food",                "cheese_slicer",
             5,  "Tom's Car Company",                  "oils",                       "cars",                  "automobiles",
```

Comment:
Looks good. I think it reflects the case that Tilman wants to cover. Here, company `id = 2` has no product. But now that I pay closer attention to the data, I notice that

1. there is no `company_id` with more than 1 product. 
2. many different companies have the same name. This might be useful to test one edge case but it's so prominent here (3 out of 5 companies) that suggests it's the most important feature of the dataset. 

Do you agree? If so, shall we open an issue to redesign the example dataset to make it more representative of the most common case (with maybe a few minor exceptions)?

Reply:
True ! That would be a great idea to do a better example data ! 
So just to be sure
1. One company can have the same id but different products, everything else the same 
2. I see, I think that when I created the example data I thought that one company id must be unique for each product - good call !!

Reply:
1. I think so based on my interpretation of what Tilman said and this 

> After assessing each product, all the products with the same category will be aggregated and set in relation to **all products of the company**.

-- https://2degreesinvesting.github.io/tiltIndicator/articles/tiltIndicator.html#product-sector-transition-risk-pstr

But best to confirm because it seems to be a fundamental part of how things work.

@Tilmon can you confirm that in the `companeis` dataset that we'll use to calculate the PSTR indicator can have multiple products for a single company?

In other words, which of A or B would be a better example of the `pstr_companies` dataset?

``` r
library(tibble)

# A
tibble(company_id = 1:3, products = 1:3)
#> # A tibble: 3 × 2
#>   company_id products
#>        <int>    <int>
#> 1          1        1
#> 2          2        2
#> 3          3        3

# B
tibble(company_id = c(1, 1, 2), products = c(1, 2, 1))
#> # A tibble: 3 × 2
#>   company_id products
#>        <dbl>    <dbl>
#> 1          1        1
#> 2          1        2
#> 3          2        1
```

<sup>Created on 2023-04-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Dear @maurolepore and @lindadelacombaz , 

case B is a more accurate reflection of the real data. There are three cases related to the number of products per company that should be covered: 
- company has no matched products, i.e. NA
- company has one unique matched product
- company has multiple matched products

Note that this refers to the matched products from ecoinvent. For the test data, it doesn't matter whether you consider the products as coming from EP or Ecoinvent I think. 

I will be able to share the real data for a sample set soon. Until then, please find the current status of the intermediate output with real data attached. THe test data contain companies with 1 or more reference_products. Note that sector info is still missing at this stage. Will keep you posted.

Reply:
> There are three cases related to the number of products per company that should be covered:
> company has no matched products, i.e. NA
> company has one unique matched product
> company has multiple matched products

Great to see you're now thinking in terms of tests! I'll capture this in an issue to ensure we're covering all these cases.